### PR TITLE
feat: Allowing preserving double dashes for variadic args (#229)

### LINKED ISSUE #229: arg removes all double dashes from the arguments, even after the first one
I would expect double dashes ("--") after the first one to be passed through as an argument. A simple test script:
```sh
#!/usr/bin/env -S usage bash
#USAGE min_usage_version "2"
#USAGE name "double_dash test"
#USAGE about "Test double dash behavior"
#USAGE arg "<args>..." help="Args" double_dash="automatic"

# echo args
echo "$usage_args"
```
The behavior appears the same no matter what double_dash behavior is specified. For example:
```
$ ./double_dash_test -- hello -- goodbye --
hello goodbye
```
I expect to see `hello -- goodbye --` since I would expect everything after the first double dash to be treated as an argument.

### PR DESCRIPTION
**Pull request by alcroito**
Previously, all double dash tokens were discarded when parsed.

A new `preserve` option has been added to the `double_dash` setting for arguments. When set to `preserve` on a variadic argument, any double dash tokens (`--`) encountered while parsing will be retained as part of the argument's value.

Just like for non-preserve behavior, `enable_flags` processing is disabled for preserve behavior as well. That's because any potential common flags like `--verbose` should be captured inside the variadic arg, rather than being treated as flags.

Added tests to verify the new behavior.

Fixes #229

---

> [!NOTE]
> Introduces a new `double_dash` behavior to retain `--` tokens in variadic args and updates parsing, spec, and docs accordingly.
> 
> - Adds `SpecDoubleDashChoices::Preserve` and serializes `double_dash` when `automatic` or `preserve`
> - Parser: `--` always disables flag parsing; when next arg is variadic with `double_dash=preserve`, the `--` token is kept as an arg value
> - Builder: `SpecArgBuilder::double_dash(...)` to set behavior programmatically
> - Docs: updates `arg.md` with `double_dash="preserve"` example
> - Tests: new cases covering preserved/skipped `--`, multiple args, and edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f1ac35b1b552bdada28d6c08ce029d92ca2ae139. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by jdx** [comment]
bugbot run

File: lib/src/spec/arg.rs [hunk 1]
```diff
@@ -21,6 +21,8 @@ pub enum SpecDoubleDashChoices {
     Optional,
     /// Require "--" to be passed
     Required,
+    /// Preserve "--" tokens as values (only for variadic args)
+    Preserve,
```
[the remark this answers is not in the document]
Comment (alcroito) [reply lib/src/spec/arg.rs]: Went with the second option.
